src/server/routes/unitReadings.js

```diff
@@ -494,25 +486,80 @@ function createRouter() {
 		if (!(validateGroupThreeDReadingsParams(req.params) && validateThreeDQueryParams(req.query))) {
 			res.sendStatus(400);
 		} else {
-			// Get time range to validate 1 year or less.
+			// Get time range - auto-adjustment handled on frontend
 			const timeInterval = TimeInterval.fromString(req.query.timeInterval);
 			if (!timeInterval.getIsBounded()) {
 				// Cannot do if not bounded.
 				res.sendStatus(400);
 			} else {
-				const duration = moment.duration(timeInterval.endTimestamp.diff(timeInterval.startTimestamp));
-				// Gets 0 unless one day beyond a year but that okay since don't do partial days.
-				const durationInYears = duration.years();
-				if (durationInYears >= 1) {
-					// Limit 3D to one year of data.
-					res.sendStatus(400);
+				const groupID = req.params.group_id;
+				const graphicUnitID = req.query.graphicUnitId;
+				const readingInterval = req.query.readingInterval;
+				const forJson = await groupThreeDReadings(groupID, graphicUnitID, timeInterval, readingInterval);
+				res.json(forJson);
+			}
+		}
+	});
+
+	// Route for getting data range for meter/group (for 3D auto-adjustment)
+	router.get('/dataRange/meters/:meter_ids', optionalAuthMiddleware, async (req, res) => {
+		if (!validateMeterLineReadingsParams(req.params)) {
+			res.sendStatus(400);
+		} else {
+			const meterIDs = req.params.meter_ids.split(',').map(idStr => Number(idStr));
+			const conn = getConnection();
+			try {
+				const result = await conn.oneOrNone(`
+					SELECT 
+						MIN(start_timestamp) as min_date,
+						MAX(end_timestamp) as max_date
+					FROM readings 
+					WHERE meter_id = ANY($1)
```

Comment:
I was thinking that the limit would be for each meter and now across all meters (or groups) since you can only graph one at a time. It makes me wonder if this route and related code should only be for a single data source. I'm not sure how that would impact the code.